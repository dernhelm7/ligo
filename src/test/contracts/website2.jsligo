/* IF YOU CHANGE THIS, CHANGE THE EXAMPLE ON THE FRONT PAGE OF THE WEBSITE */

type storage = int;

/* variant defining pseudo multi-entrypoint actions */

type Increment = {
  kind: 'increment',
  value: int
}

type Decrement = {
  kind: 'decrement',
  value: int
}

type parameter =
| Increment
| Decrement;

let add = ([a,b]: [int, int]): int => a + b;
let sub = ([a,b]: [int, int]): int => a - b;

/* real entrypoint that re-routes the flow based on the parameter provided */

let main = ([p,storage]: [parameter, storage]) => {
  switch (p.kind) {
    case 'increment':
      let storage2 = add ([storage, p.value]);
      return [[] as list <operation>, storage2];
    case 'decrement':
      let storage3 = sub ([storage, p.value]);
      return [[] as list <operation>, storage3];
  }  
};

/* IF YOU CHANGE THIS, CHANGE THE EXAMPLE ON THE FRONT PAGE OF THE WEBSITE */
