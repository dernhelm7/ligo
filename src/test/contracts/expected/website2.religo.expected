type storage = int;

type parameter = Increment(int) | Decrement(int);

let add = ((a, b): (int, int)): int => a + b;

let sub = ((a, b): (int, int)): int => a - b;

let main = ((p, storage): (parameter, storage)) => {
  let storage = 
    switch(p){
    | Increment(n) => add((storage, n))
    | Decrement(n) => sub((storage, n))
    };
  ([] : list(operation), storage)
};
